raiden_libs/network/matrix/client.py

```diff
@@ -259,6 +274,7 @@ def _mkroom(self, room_id: str) -> Room:
 
     def stop_listener_thread(self):
         """ Kills sync_thread greenlet before joining it """
+        self.should_listen = False
         if self.sync_thread:
             self.sync_thread.kill()
```

Comment:
`kill()`? you savage!
I'd change `should_listen` to `Event` and join it as creators of gevent intendend

Reply:
`kill` should only **at most** kill the long-polling connection to `/sync` (can get stuck there waiting for some server event for like 30s), which on my tests made the connection raise a ConnectionError because _the remote end closed the connection without a response_, which is untrue, **we** closed the connection, but that's expected, and handled in the `_sync` api call and suppressed when `should_listen` is False. About it, it should remain as a boolean, as it is used and depended on in the parent class (`MatrixClient`), so this boolean set here is just going with the interface, to ensure the `/sync` call in `_sync` will exit the greenlet gracefully.

Reply:
ok, got it. you could maybe put some comments in there, it'd be useful to have some explanation how does the sync work.